### ISSUE TITLE
Using LightFM for explicit interaction matrix

### ISSUE THREAD

**Issue by SantGIIT** [lyst/lightfm#299]
Hi, I'm new to recommendation system so a silly question. 

I have user-item matrix with binary values (if a user purchased the item, the value is 1 otherwise 0). I choose 80% of the known entry as training set and after converting this to sparse matrix I trained the model. With the 20% test-set the precision@10 comes very low (almost zero) and AUC comes near about 50%, i.e., almost the result similar to random guess. Where I'm wrong? Please, help. Thanks.

**Comment by maciejkula** [comment]
Many things could be wrong, but most of the time it's either (1) data errors or (2) bad hyperparameters.

What are your hyperparameters and what is the shape and density of your data?

**Comment by SantGIIT** [comment]
Sorry that I forgot to mention that as of now, I'm experimenting with collaborative filtering only, i.e., I haven't specified the user and item feature matrix (but tried specifying those as identity matrix too).
 
Hyperparameters: learning_rate=0.01, loss='bpr', no_components=10, epoch=20.

In a typical example,  the number of users: 1500, the number of items: 1000, Sparsity: 1.694% and the rating matrix is binary, so couldn't understand what do you mean by the distribution of the data. 

Performance: 
 Train set size: 17411
 Test set size: 4353
 Training model WITHOUT meta-info.
 Item Embedding Matrix Dims: (1000, 10)
 User Embedding Matrix Dims: (1500, 10)
 Precision: train 0.02, test 0.01.
 AUC: train 0.54, test 0.53.

**Comment by maciejkula** [comment]
Your hyperparameters look OK, maybe you could try a higher learning rate and larger `no_components`?

**Comment by SantGIIT** [comment]
With loss='logistic', and learning_rate=0.05, no_components=30, the result is
 Precision: train 0.00, test 0.00.
 AUC: train 0.82, test 0.23.

**Comment by maciejkula** [comment]
That does look better. In general, you always want to do a hyperparameter search. Have a look [here](https://gist.github.com/maciejkula/29aaf2db2efee5775a7f14dc387f0c0f) for some sample code.

**Comment by SantGIIT** [comment]
Hi Maciej,

Thanks for the help. Have got improved performance in CF without content
info.  in terms of AUC. But, now I have another difficulty figuring out why
incorporating the item's content information is not helping in my case.
Here is what the performance my code reported.

 Loading the data...
 Creating Interaction Matrix
 Created Size: (5000, 3)
=====================================
 Starting likes info
 Number of users: 1686
 Number of items: 4381
 Sparsity: 0.068%
 Ending likes info
 Number of users: 1678
 Number of items: 3903
=====================================
Tuning parameters without metadata
=====================================
Without Metainfo Best score 0.535032510757 at {'num_epochs': 49,
'no_components': 35, 'user_alpha': 4.7024178561511364e-09, 'max_sampled':
10, 'item_alpha': 2.326188727461577e-09, 'loss': 'bpr', 'learning_rate':
0.019314686949204716, 'learning_schedule': 'adadelta'}
=====================================
 Training model WITHOUT meta-info.
=====================================
 Item Embedding Matrix Dims: (3903, 35)
 User Embedding Matrix Dims: (1678, 35)
 Precision without Meta-info: *train 0.12, test 0.00.*
 AUC without Meta-info: *train 1.00, test 0.53.*

=====================================
Tuning parameters with metadata
=====================================
With Metainfo Best score 0.528344392776 at {'num_epochs': 41,
'no_components': 45, 'user_alpha': 3.792675779791911e-09, 'max_sampled':
11, 'item_alpha': 1.2647070730452016e-09, 'loss': 'bpr', 'learning_rate':
0.01985679207599666, 'learning_schedule': 'adadelta'}
=====================================
 Training model WITH meta-info.
=====================================
Precision with meta-info: *train 0.00, test 0.00.*
AUC with meta-info:* train 0.54, test 0.50.*
Item Embedding Matrix Dims: (24, 45)
User Embedding Matrix Dims: (37, 45)


I understand that data is very sparse in this example, but I have increased
sparsity level to 5% but still no improvement.
Please, help.

Santosh

**Comment by SimonCW** [comment]
I’m closing this issue because it has been inactive for a long time. If you still encounter the problem, please open a new issue.

Thank you!
